python_wrapper/cap.py

```diff
@@ -0,0 +1,417 @@
+                          "parties", "judges", "attorneys", "citations", "url", "head", "body"]
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+            
+            while True:
+                for case in current_page["results"]:
+                    case_data = {
+                        "id": case["id"],
+                        "name": case["name"],
+                        "name_abbreviation": case["name_abbreviation"],
+                        "decision_date": case["decision_date"],
+                        "court_id": case["court"]["id"],
+                        "court_name": case["court"]["name"],
+                        "court_slug": case["court"]["slug"],
+                        "parties": str(case["casebody"]["data"]["parties"]),
+                        "judges": str(case["casebody"]["data"]["judges"]),
+                        "attorneys": str(case["casebody"]["data"]["attorneys"]),
+                        "citations": str(case["citations"]),
+                        "url": case["url"],
+                        "head": case["casebody"]["data"]["head_matter"],
+                        "body": case["casebody"]["data"]["opinions"][0]["text"]
+                    }     
+                    writer.writerow(case_data)
+            
+                try:
+                    next_result = self._request(current_page["next"])
+                    current_page = next_result.json()
+                    
+                except:
+                    break
+        
+        print("Downloaded " + str(search_results["count"]) + " court cases to file " + filename + ".") 
+        
+        
+    def download_mltpl_courts(self, search_results, filename):
+        """
+        Use for downloading  
+        Input a JSON list of URI search results from multi_search_cases (full_text MUST be set to true) and 
+        downloads the search results as a .csv file. 
+        
+        :param search_results: JSON search result retrieved using the 'search_cases' method that you wish to
+                                download
+        :type search_results: JSON
+        :param filename: desired filename of downloaded data
+        :type filename: str
+        
+        :return: null
+        """
+    
+        court_index = 0
+        row_count = 0
+        search_results = search_results
+        current_uri = search_results[court_index]
+        
+        downloaded_count = 0
+        
+        with open(filename, "w", encoding='utf-8') as csvfile:
+            fieldnames = ["id", "name", "name_abbreviation", "decision_date", "court_id", "court_name", 
+                          "court_slug", "parties", "judges", "attorneys", "citations", "url", "head", "body"]
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+            
+            current_page = self._request(current_uri).json()
+
+            while True:
+                try:
+                    if current_page["count"] > 0:
```

Comment:
if there are too many results, we skip counting, otherwise the response would take too long to fulfill. This will error out and skip writing rows. I think a better check is `if current_page["count"] != 0`